### PR TITLE
Beacon typescript annotations

### PR DESCRIPTION
**Pull request by gsfk**
Converts all beacon code to Typescript. Also: 

- update deprecated redux-toolkit code
- misc fixes, mostly issues unearthed by typing

### CODE REVIEW

File: src/js/components/Beacon/BeaconSearchResults.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { useAppSelector } from '@/hooks';
+import SearchResultsPane from '../Search/SearchResultsPane';
+
+const BeaconSearchResults = () => {
+  const isFetchingData = useAppSelector((state) => state.beaconQuery?.isFetchingQueryResponse);
```
Comment (davidlougheed) [inline src/js/components/Beacon/BeaconSearchResults.tsx:6]: dont need `?` in the `beaconQuery?` here I think - it should always be an object (and if it isn't somehow, the line below can error)

File: src/js/features/beacon/beaconConfig.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
```
Comment (davidlougheed) [inline src/js/features/beacon/beaconConfig.ts:1]: rename to match other ts files in `features/`

File: src/js/features/beacon/beaconQuery.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
```
Comment (davidlougheed) [inline src/js/features/beacon/beaconQuery.ts:1]: rename to match other `features/` files

**Review by davidlougheed — CHANGES_REQUESTED** [review]

File: src/js/components/Beacon/BeaconSearchResults.tsx (hunk 1 above)
Comment (gsfk) [reply src/js/components/Beacon/BeaconSearchResults.tsx]: indeed yes

**Review by davidlougheed — APPROVED** [review]
lgtm. unfortunately for the translation pr...

**Review by SanjeevLakhwani — APPROVED** [review]
lgtm

File: src/js/components/Beacon/Filter.tsx [hunk 4]
```diff
@@ -57,22 +62,25 @@ const Filter = ({ filter, form, querySections, removeFilter, isRequired }: Filte
...
+          placeholder={td('select a search field')}
           style={FILTER_FORM_ITEM_INNER_STYLE}
           onSelect={handleSelectKey}
           options={searchKeyOptions(querySections)}
         />
       </Form.Item>
       <Form.Item
         name={`filterValue${filter.index}`}
-        rules={[{ required: isRequired, message: 'value required' }]}
+        rules={[{ required: isRequired, message: td('value required') as string }]}
```
Comment (SanjeevLakhwani) [inline src/js/components/Beacon/Filter.tsx:77]: There exists a type enforcing hook for this, use that

**Review by davidlougheed — APPROVED** [review]
lgtm

**Pull request merged**